dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/ServerTransportObserver.java

```diff
@@ -48,19 +59,38 @@ public void onMetadata(Metadata metadata, boolean endStream) {
             headers.set(TripleHeaderEnum.CONTENT_TYPE_KEY.getHeader(), TripleConstant.CONTENT_PROTO);
         }
         ctx.writeAndFlush(new DefaultHttp2HeadersFrame(headers, endStream))
-            .addListener(future -> {
-                if (!future.isSuccess()) {
-                    promise.tryFailure(future.cause());
-                }
-            });
+                .addListener(future -> {
```

Comment:
why discard `promise` here ?

Reply:
This PR has an explanation.
https://github.com/apache/dubbo/pull/8922